### ISSUE TITLE
Trainning more epochs seems to perform worse

### ISSUE THREAD

**Issue by FranciscoPark** [AkaCoder404/TennisCourtDetectorwithYolov8#1]
Hey, nice work you did. I was using your codes to train as well. I tried 10 epochs just like you did but some points seem to be a little off. So I tried 20 epochs which made it significantly worse. Do you have any clue why this is happening? 
![bMMueLic82A_1950](https://github.com/AkaCoder404/TennisCourtDetectorwithYolov8/assets/76905169/05a8e92f-045d-4561-9157-ccb71741fb5b)

**Comment by FranciscoPark** [comment]
This is the result of running it for 20 epochs. 
![results(1)](https://github.com/AkaCoder404/TennisCourtDetectorwithYolov8/assets/76905169/795d466d-6b87-43eb-8cd5-368a8cc8cf61)

**Comment by kaikai23** [comment]
Hi @FranciscoPark ,
I also trained this repo with 10, 20 epochs, and I got the same observation as yours. I found that it is because the `mosaic` parameter. If you train with 20 epochs, you should use 
```
model.train(data="datasets_config/tennis.yml",
            epochs=args.epochs,
            close_mosaic=args.epochs,
            imgsz=args.imgsz,
            plots=True,
            save=args.save_model)
``` 
so that mosaic is disabled from the first ever epoch.

**Comment by kaikai23** [comment]
@FranciscoPark  @AkaCoder404  I find that no matter I train with 10 or 20 epochs, the precision and recall for pose (P) are always 0 in validation, although the val visualizations look ok. Do you have any insights about this?

I would need to train with some other court and cumtom keypoints, so it's important to have a quantitative val metric to show the model performance ...

**Comment by kaikai23** [comment]
Ok, I found the bug. In `convert_to_yolo.ipynb` when converting val keypoints, there is one line of code missing:
`x, y = kp`
After adding this line of code, the pose metrics become normal.

**Comment by AkaCoder404** [comment]
Nice find!
